apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/GrpcRdaSource.java

```diff
@@ -363,6 +394,27 @@ public Duration getMaxIdle() {
     private ManagedChannelBuilder<?> createInProcessChannelBuilder() {
       return InProcessChannelBuilder.forName(inProcessServerName);
     }
+
+    private Long parseJWTExpirationDate(String token) {
+      try {
+        String[] jwtBits = token.split("\\.");
+        String claimsString = new String(Base64.getDecoder().decode(jwtBits[1]));
+        JWTClaims claims = mapper.readValue(claimsString, JWTClaims.class);
+        if (claims.exp == null) throw new NullPointerException();
+        return claims.exp;
+      } catch (NullPointerException e) {
+        LOGGER.error("Could not find expiration claim", e);
```

Comment:
I think both 406 and 408 can be warn statements, what do you think?

Reply:
It seemed to be the case that `warn` logs do not actively notify anyone of the issue.  Being that this check will only happen when the service first starts, it seemed like a good idea to throw an error in this case to alert someone immediately that something is wrong.

Reply:
Yes, currently there are no cloudwatch alarms setup to monitor WARN statements (although we could add that if we wanted to).  My thought here is we should trigger some sort of alarm/error if either 
  a) the token doesn't allow us to connect to the RDA API
  b) we believe the token is going to fail in the near future (in this case because the exp time is approaching)
  
Any other issue with the JWT may be a false negative, in which case I'm not sure an ERROR statement is appropriate.  Take the following scenario, for example

Let's say MPSM stops putting the `exp` key in the JWT because they want their tokens to live forever and issues a new one for the BFD team to use.  Someone on the BFD receives that token and updates it in the configuration.  Once deployed, this code would start issuing logging messages saying it can't find the expiration claim, but that's totally fine, as the MPSM team has changed their strategy on issuing JWTs.  I think a WARN is fine, I would just hate for someone to be paged about this when it's not an ERROR it's just a strategy shift on how MPSM wants to handle authentication.